src/backend/distributed/planner/multi_logical_planner.c

```diff
@@ -997,6 +997,23 @@ TargetListOnPartitionColumn(Query *query, List *targetEntryList)
 		Expr *targetExpression = targetEntry->expr;
 
 		bool isPartitionColumn = IsPartitionColumn(targetExpression, query);
+		Oid relationId = InvalidOid;
```

Comment:
Otherwise I need to re-implement outer if block given in the `IsPartitionColumn` function.